core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -878,19 +888,28 @@ private[kafka] class Processor(val id: Int,
 
 class ConnectionQuotas(val defaultMax: Int, overrideQuotas: Map[String, Int]) {
 
-  private val overrides = overrideQuotas.map { case (host, count) => (InetAddress.getByName(host), count) }
+  private var maxConnectionsPerIp = defaultMax
+  private var overrides = overrideQuotas.map { case (host, count) => (InetAddress.getByName(host), count) }
```

Comment:
nit: can we just a more descriptive name? Maybe it should be `defaultMaxConnectionsPerIp` and `maxConnectionsPerIpOverrides` or something like that?